geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSUnionIntegrationTest.java

```diff
@@ -56,31 +63,122 @@ public void sunionErrors_givenTooFewArguments() {
+  @Test
+  public void sunionWithSetAndNonExistentSet_returnsAllValuesInSet() {
+    jedis.sadd(setKey1, setMembers1);
+    assertThat(jedis.sunion(setKey1, nonExistentSetKey))
+        .containsExactlyInAnyOrder(setMembers1);
+  }
 
-    jedis.sadd("{user1}newEmpty", "born2die");
-    jedis.srem("{user1}newEmpty", "born2die");
-    Set<String> yetAnotherResultSet = jedis.sunion("{user1}set2", "{user1}newEmpty");
-    assertThat(yetAnotherResultSet).containsExactlyInAnyOrder(secondSet);
+  @Test
+  public void sunionWithMultipleNonExistentSets_returnsEmptySet() {
+    assertThat(jedis.sunion("{user1}nonExistentSet1", "{user1}nonExistentSet2")).isEmpty();
+  }
+
+  @Test
+  public void sunionWithNonOverlappingSets_returnsUnionOfSets() {
+    String[] secondSetMembers = new String[] {"apple", "microsoft", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String[] result =
+        {"one", "two", "three", "four", "five", "apple", "microsoft", "linux", "peach"};
+    assertThat(jedis.sunion(setKey1, setKey2)).containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionWithSetsWithSomeSharedValues_returnsUnionOfSets() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+
+    String[] result = {"one", "two", "three", "four", "five", "linux", "peach"};
+    assertThat(jedis.sunion(setKey1, setKey2)).containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionWithSetsWithAllSharedValues_returnsUnionOfSets() {
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, setMembers1);
+
+    assertThat(jedis.sunion(setKey1, setKey2)).containsExactlyInAnyOrder(setMembers1);
+  }
+
+  @Test
+  public void sunionWithMultipleSets_returnsUnionOfSets() {
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    String thirdSetKey = "{user1}setKey3";
+    String[] thirdSetMembers = new String[] {"fire", "dance", "floor", "five"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKey2, secondSetMembers);
+    jedis.sadd(thirdSetKey, thirdSetMembers);
+
+
+    String[] result =
+        {"one", "two", "three", "four", "five", "linux", "peach", "fire", "dance", "floor"};
+    assertThat(jedis.sunion(setKey1, setKey2, thirdSetKey))
+        .containsExactlyInAnyOrder(result);
+  }
+
+  @Test
+  public void sunionWithSetsFromDifferentSlots_returnsCrossSlotError() {
+    String setKeyDifferentSlot = "{user2}setKey2";
+    String[] secondSetMembers = new String[] {"one", "two", "linux", "peach"};
+    jedis.sadd(setKey1, setMembers1);
+    jedis.sadd(setKeyDifferentSlot, secondSetMembers);
+
+    assertThatThrownBy(() -> jedis.sunion(setKey1, setKeyDifferentSlot))
+        .hasMessageContaining(ERROR_DIFFERENT_SLOTS);
+  }
+
+
+  @Test
+  public void sunionWithDifferentKeyType_returnsWrongTypeError() {
```

Comment:
Could we also get a test where the first key passed to SUNION is a set, but the second key is some other type please?